test/views/declaration/CarrierEoriNumberViewSpec.scala

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.declaration
+
+import base.Injector
+import forms.common.Eori
+import forms.common.YesNoAnswer.YesNoAnswers
+import forms.declaration.carrier.CarrierEoriNumber
+import models.{DeclarationType, Mode}
+import models.requests.JourneyRequest
+import org.jsoup.nodes.Document
+import org.scalatest.Matchers._
+import play.api.data.Form
+import services.cache.ExportsTestData
+import unit.tools.Stubs
+import views.components.gds.Styles
+import views.declaration.spec.UnitViewSpec
+import views.html.declaration.carrier_eori_number
+import views.tags.ViewTest
+
+@ViewTest
+class CarrierEoriNumberViewSpec extends UnitViewSpec with ExportsTestData with Stubs with Injector {
+
+  private val page: carrier_eori_number = instanceOf[carrier_eori_number]
+
+  private def createView(mode: Mode = Mode.Normal, form: Form[CarrierEoriNumber] = CarrierEoriNumber.form())(
+    implicit request: JourneyRequest[_]
+  ): Document =
+    page(mode, form)(request, messages)
+
+  "Carrier Eori Number View" should {
+    onJourney(DeclarationType.CLEARANCE) { implicit request =>
```

Comment:
Should there not be `onEveryDeclarationJourney` instead? Right now, you are testing it against a single journey only.

Reply:
Fixed. Added all journey types apart from SUPPLEMENTAL (that does not have Carrier in its journey flow)